asciidoc/courses/llm-fundamentals/modules/2-vectors-semantic-search/lessons/2-vector-index/lesson.adoc

```diff
@@ -0,0 +1,222 @@
+
+
+[TIP]
+.Generating the Embeddings
+====
+We wrote a simple Python script to call the `/api/embeddings` endpoint served by Ollama.
+You can link:TODO:link-the-repo[view the code on Github^].
+====
+
+For this lesson, it doesn't necessarily matter where these embeddings were obtained as each LLM will provide an embedding in its own shape.
+
+
+== Creating the Vector Index
+
+For simplicity, let's say the embeddings will be saved to the `.embedding` property on the `(:Movie)` node.
+We must create a vector index on that Label and property
+
+
+As vector indexes are a specialized type of index, you must call the `db.index.vector.createNodeIndex()` procedure.
+The procedure expects five parameters.
+
+.Procedure Signature
+[source,cypher]
+----
+CALL db.index.vector.createNodeIndex(
+    'moviePlots', // <1>
+    'Movie',      // <2>
+    'embedding',  // <3>
+    3200,         // <4>
+    'cosine'      // <5>
+)
+----
+
+1. The name of the index - we will want to call ours something `moviePlots`.
+2. The label on which to index - `Movie`.
+3. The key of the property on the node to index - `embedding`.
+4. The dimension of the embedding - Orca Mini embeddings consist of `3200` dimensions.
+5. The similarity function to use when comparing values in this index - this can be `euclidean` or `cosine`.
+
+[TIP]
+.Choosing a Similarity Function
+====
+You will want to choose the similarity function that is closest to the loss function used for training the embedding model.
+You may be able to find this out by reviewing your model provider's documentation.
+
+Experiment with the functions if necessary, and, if all else fails, experiment with `cosine` first.
+
+You can link:https://neo4j.com/docs/cypher-manual/current/indexes-for-vector-search/#indexes-vector-similarity[read more about similarity functions in our documentation^].
+====
+
+
+
+=== Check the index creation status
+
+You can check that the index has been created using the `SHOW INDEXES` command.
+
+.Show Indexes
+[source,cypher]
+----
+SHOW INDEXES  YIELD id, name, type, state, populationPercent WHERE type = "VECTOR"
+----
+
+.Show Indexes Result
+|===
+| id | name | type | state | populationPercent
+
+|1 | "moviePlots" | "VECTOR" | "ONLINE" | `100.0`
+|===
+
+Once the `state` is listed as online, the index will be ready to query.
+The `populationPercentage` field indicates the proportion of node and property pairing that currently exist in the index.
+
+
+== Loading Embeddings
+
+You can load these embeddings into the Neo4j Sandbox instance using the `LOAD CSV` command.
+
+The following data loads a CSV file hosted on Github, performs a `MATCH` query to find the `(:Movie)` node with the corresponding `movieId` property, and then sets the `.embedding` property on that node.
+
+
+
+.Loading the Embeddings
+[source,cypher]
+----
+LOAD CSV WITH HEADERS
+FROM 'https://github.com/neo4j-graphacademy/llm-fundamentals/raw/main/data/embeddings.csv'
```

Comment:
perhaps use the github.io CDN or data.neo4j.com instead? GH can sometimes be a bit slow?